Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -819,6 +750,46 @@ WTFLogChannel& FullscreenManager::logChannel() const
 }
 #endif
 
+FullscreenManager::FullscreenPromise::~FullscreenPromise()
+{
+    clear();
+}
+
+auto FullscreenManager::FullscreenPromise::operator=(CompletionHandler<void(ExceptionOr<void>)>&& promise) -> FullscreenPromise&
```

Comment:
The name of this class is weird, as is the name of the parameter, and the ivar. None of them is a promise; the parameter and the ivar are both `CompletionHandler`s, not Promises.

Reply:
I agree it's a strange name.  It rejects and resolves like a promise, and it often wraps a promise.  It'll hopefully be soon removed.